### PR TITLE
Add new game event: CC Enemy communication interception

### PR DESCRIPTION
**Pull request by VasilisThePikachu**
## About the PR

This ports over an event from tg station. That raises the security level from Green to Blue

The pause at the start of the audio file is intentional. Lets the little beep boop to shine through.

Idk how "Weight" works its undocumented so I just put 10

## Why / Balance

I think that it can add something to keep players on their feet if they have been on green for a while. I can also see some roleplay going into this about these "communications"

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

https://github.com/space-wizards/space-station-14/assets/34938708/925c9a89-eb42-402a-8126-277d2d2a287e

## Breaking changes

**Changelog**

:cl: Vasilis
- add: Added a new game event. Central command may raise your alert level to blue randomly to keep you on your feet.

### CODE REVIEW

File: Content.Server/StationEvents/Components/CommunicationInterceptionRuleComponent.cs
Comment (deltanedas) [inline Content.Server/StationEvents/Components/CommunicationInterceptionRuleComponent.cs:12]: remove readonly and add [DataField]
also im pretty sure this can use ProtoId<AlertLevelPrototype> or something

File: Content.Server/StationEvents/Events/CommunicationInterceptionRule.cs
Comment (deltanedas) [inline Content.Server/StationEvents/Events/CommunicationInterceptionRule.cs:30]: there should be some actual effect since everyone will know its le event and go back to green

**Review by deltanedas — CHANGES_REQUESTED** [review]

File: Content.Server/StationEvents/Events/CommunicationInterceptionRule.cs
Comment (Emisse) [reply Content.Server/StationEvents/Events/CommunicationInterceptionRule.cs:30]: agree, maybe reconfigure this to sleeper agents

Comment (VasilisThePikachu) [reply Content.Server/StationEvents/Events/CommunicationInterceptionRule.cs:30]: the frick is a sleeper agent

Comment (deltanedas) [reply Content.Server/StationEvents/Events/CommunicationInterceptionRule.cs:30]: make a random player tot :trollface:

**Comment by VasilisThePikachu** [comment]
Making into a pr until #23445 is merged

this will do sleeper agents

https://discord.com/channels/310555209753690112/770682801607278632/1194779701454053477 little discussion (delta join discord)

**Comment by deltanedas** [comment]
no noob :trollface:

**Comment by VasilisThePikachu** [comment]
You will get free nitro and maintainer if you do (pinky promise)

**Comment by K-Dynamic** [comment]
> Making into a draft until #23445 is merged
> 
> this will do sleeper agents
> 
> https://discord.com/channels/310555209753690112/770682801607278632/1194779701454053477 little discussion (delta join discord)

Sleeper agents (midround syndies) real

Break the meta please

**Comment by Rainfey** [comment]
Midround syndie's kinda already exist.
If the number of traitors is below a certain number (based on player count) - then everytime a player joins there is a chance to make that player a traitor

You can see the relevant code here:
https://github.com/space-wizards/space-station-14/blob/064d52db412d9cf80ace425db8573aea2dfc7d81/Content.Server/GameTicking/Rules/TraitorRuleSystem.cs#L272

**Comment by VasilisThePikachu** [comment]
> Midround syndie's kinda already exist. If the number of traitors is below a certain number (based on player count) - then everytime a player joins there is a chance to make that player a traitor
> 
> You can see the relevant code here:
> 
> https://github.com/space-wizards/space-station-14/blob/064d52db412d9cf80ace425db8573aea2dfc7d81/Content.Server/GameTicking/Rules/TraitorRuleSystem.cs#L272
> 
> This of course only gets triggered if the TraitorGameRule is active.

We are aware of this. I brought it up

HOWEVER

Lets say you were not a syndie for like 30 mins and you managed to get the trust of a lot of people. Confirmed by sec to be not a syndie etc. Now suddenly you are made one

Thats an advantage to you. Makes the round intresting

**Comment by VektorZ1** [comment]
Send a random alert from past rounds on behalf of a random head with this :trollface:

**Comment by Everturning** [comment]
sleeper agents dont get uplink jsyk

**Comment by deltanedas** [comment]
no.uplink then its just legal selfantag :trollface:

**Comment by K-Dynamic** [comment]
> with no uplink then its just legal selfantag :trollface:

Kinda but they'd still have objectives rather than just murderboning (even if it's silly like HoP needing Ian meat, which they proudly shared with other syndies as an admin-forced syndie)

**Comment by deltanedas** [comment]
yes mr sleeper agent you will get disk (which is already impossible long term without just bombing evac like a bitch and stealing caps bag right before it docks) with a grand total of 0 (zero) tk :trollface:

**Comment by Everturning** [comment]
in 13 they have no uplink, why should they have it for us?

**Comment by VasilisThePikachu** [comment]
> in 13 they have no uplink, why should they have it for us?

Cause we are not ss13. We decide stuff like this.

**Comment by deltanedas** [comment]
according to latest livrah video on terry they have 20tc, snopes factcheck: fake

**Comment by Everturning** [comment]
is work still being done on this?

**Comment by VasilisThePikachu** [comment]
> is work still being done on this?

This is pending the antag refactor

**Comment by VasilisThePikachu** [comment]
Antag refactor has been merged. RIGHT on the day I go on vacation...

**Comment by metalgearsloth** [comment]
Re-open when you work on it.

**Comment by Dutch-VanDerLinde** [comment]
can i work on this

**Comment by VasilisThePikachu** [comment]
> can i work on this
Go ahead, i don't have an interest on this anymore. I need fewer projects

**Pull request closed without merging**
